apps/storefront-e2e/src/support/page_fragments/header.fragment.ts

```diff
@@ -34,30 +34,39 @@ export class HeaderFragment {
   };
 
   changeLocale = (locale: string) => {
-    cy.intercept({
-      method: 'GET',
-      url: /(\/catalog-search.*|\/concrete-products\/.*)/,
-    }).as('productOrSearchRequests');
-    // hydrate
-    this.getLocaleButton().click();
-    // eslint-disable-next-line cypress/no-unnecessary-waiting
-    cy.wait(500);
-    this.getLocaleButton().click();
-    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    this.initProductsUpdateInterceptor();
+
+    this.getLocaleButton().trigger('mouseenter').wait(75).click();
```

Comment:
What is the trick - why we should trigger mouse enter first and wait a bit and then click on it?

Reply:
The `mouseenter` event triggers component hydration 

The wait is needed to postpone a click on the element while `js` file is being loaded and the component is re-rendered

I've tried to solve it in a better way (without waiting) by checking the `defer-hydration` attribute existence on the element - but it didn't work. Let's have it as it is now, and when Stan is back maybe we will try to improve it further